lib/fetch/request.js

```diff
@@ -467,8 +467,8 @@ class Request {
       // not contain `Content-Type`, then append `Content-Type`/Content-Type to
       // this’s headers.
       if (contentType && !this[kHeaders].has('content-type')) {
-        this[kHeaders].append('content-type', contentType)
-        this[kState].headersList.append('content-type', contentType)
+        this[kHeaders].set('content-type', contentType)
+        this[kState].headersList.set('content-type', contentType)
```

Comment:
I think the reason for the duplication is this[kHeaders][kHeadersList] === this[kState].headersList, making this second call redundant.

See
https://github.com/nodejs/undici/blob/0babdb1371085ff5b1f3fcec28256df2384fb177/lib/fetch/headers.js#L215

Reply:
I thought so too, but removing the second call caused some tests (from the node-fetch test suite) to fail.

Reply:
Yes, I think here we should use `.set`. If I remember correctly, the old implementation was using `.set/.push` as well.